### ISSUE TITLE
'ChargebeeReactNative-Swift.h' file not found

### ISSUE THREAD

**Issue by kylea3** [chargebee/chargebee-react-native#178]
Hi, I added "@chargebee/react-native-chargebee": "^2.4.5" to my project and now all builds fail with the the following error:

node_modules/@chargebee/react-native-chargebee/ios/ChargebeeReactNative.mm:2:9)

  1 | #import "ChargebeeReactNative.h"
> 2 | #import <ChargebeeReactNative-Swift.h>
    |         ^ 'ChargebeeReactNative-Swift.h' file not found
  3 | 
  4 | @implementation ChargebeeReactNative
  5 | RCT_EXPORT_MODULE()
The following build commands failed:
▸ 	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/Tallio-bbielmlrqyffaodyqcvcjhdlhgly/Build/Intermediates.noindex/ArchiveIntermediates/Tallio/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/ChargebeeReactNative.build/Objects-normal/arm64/ChargebeeReactNative.o /Users/expo/workingdir/build/node_modules/@chargebee/react-native-chargebee/ios/ChargebeeReactNative.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ChargebeeReactNative' from project 'Pods')
▸ (1 failure)
** ARCHIVE FAILED **
The following build commands failed:
	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/Tallio-bbielmlrqyffaodyqcvcjhdlhgly/Build/Intermediates.noindex/ArchiveIntermediates/Tallio/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/ChargebeeReactNative.build/Objects-normal/arm64/ChargebeeReactNative.o /Users/expo/workingdir/build/node_modules/@chargebee/react-native-chargebee/ios/ChargebeeReactNative.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ChargebeeReactNative' from project 'Pods')
(1 failure)

Adding @chargebee/react-native-chargebee to a new project doesn't cause and failed builds, so I am wondering if this has to do with another library that I imported for my project. One of the other libraries I am using (@react-native-firebase/app) uses static frameworks. Here is the implementation instructions from that package:

Open the file ./ios/Podfile and add this line inside your targets (right before the use_react_native line in current react-native releases that calls the react native Podfile function to get the native modules config):

use_frameworks! :linkage => :static
To use Static Frameworks on iOS, you also need to manually enable this for the project with the following global to your /ios/Podfile file:

# right after `use_frameworks! :linkage => :static`
$RNFirebaseAsStaticFramework = true

Would static frameworks be causing the compatibility issues? If not, have you ran into this problem in any other scenarios?

**Comment by FontSanguinetti** [comment]
Same problem here. On android it works just fine but in iOS in throws this same error log: 'ChargebeeReactNative-Swift.h' file not found

**Comment by kylea3** [comment]
@cb-haripriyan I have tried setting it to static, but I still get the same error. I've spent the last couple of weeks trying to solve for this issue but haven't found anything. I figured it had to be something with my build settings, but everything seems to be in order.

**Comment by cb-haripriyan** [comment]
@kylea3 Seems to be an issue when linking the SDK as a static framework rather than a static library.

**Comment by Brayyan-Yanes** [comment]
Hi!, something like that in your podfile can help you to fix as well.

pre_install do |installer|
  $dynamic_frameworks = [
    'ChargebeeReactNative',
  ]
  installer.pod_targets.each do |pod|
    if $dynamic_frameworks.include?(pod.name)
      puts "Overriding the build_type method for #{pod.name}"
      def pod.build_type;
        # NOTE: This syntax only works in cocoapods >= v1.11. Pre 1.11 BuildType was
        # a simple single value.
        BuildType.new(:linkage => :static)
      end
    end
  end
end

target 'YourProject' do ....

**Comment by katche70** [comment]
I have the same problem. Why is the ticket closed? What exactly is the solution?

**Comment by cb-haripriyan** [comment]
@katche70 Can you confirm if [this](https://github.com/chargebee/chargebee-react-native/issues/178#issuecomment-2082231027) didn't solve the issue for you?

**Comment by katche70** [comment]
> @katche70 Can you confirm if [this](https://github.com/chargebee/chargebee-react-native/issues/178#issuecomment-2082231027) didn't solve the issue for you?

yes, it did not solve my problem. checked this settings and it is already set as static library

**Comment by katche70** [comment]
Here is my pod install, i'm using react native with expo 

> pod install
Using Expo modules
[Expo] Enabling modular headers for pod ExpoModulesCore
[Expo] Enabling modular headers for pod React-Core
[Expo] Enabling modular headers for pod GoogleSignIn
[Expo] Enabling modular headers for pod ReactCommon
[Expo] Enabling modular headers for pod React-RCTAppDelegate
[Expo] Enabling modular headers for pod expo-dev-menu-interface
[Expo] Enabling modular headers for pod EXManifests
[Expo] Enabling modular headers for pod EXUpdatesInterface
[Expo] Enabling modular headers for pod expo-dev-menu
[Expo] Enabling modular headers for pod RCT-Folly
[Expo] Enabling modular headers for pod glog
[Expo] Enabling modular headers for pod expo-dev-launcher

Adding a custom script phase for Pod RNFBApp: [RNFB] Core Configuration
Auto-linking React Native modules for target `App`: ChargebeeReactNative, RNAppleAuthentication, RNAudioRecorderPlayer, RNCClipboard, RNCPushNotificationIOS, RNDeviceInfo, RNFBApp, RNFBMessaging, RNFastImage, RNFileViewer, RNFlashList, RNGestureHandler, RNGoogleSignin, RNKeychain, RNReactNativeHapticFeedback, RNReanimated, RNSVG, RNScreens, react-native-blob-util, react-native-charts-wrapper, react-native-menu, react-native-mmkv, react-native-netinfo, react-native-pager-view, react-native-render-html, react-native-safe-area-context, react-native-segmented-control, react-native-skia, and react-native-webview
Framework build type is static framework
[Codegen] Generating ./build/generated/ios/React-Codegen.podspec.json
[Codegen] generating an empty RCTThirdPartyFabricComponentsProvider
Analyzing dependencies
[Codegen] Found FBReactNativeSpec
[Codegen] Found rncore
Downloading dependencies
Generating Pods project
Setting USE_HERMES build settings
Setting REACT_NATIVE build settings
Setting CLANG_CXX_LANGUAGE_STANDARD to c++20 on /App.xcodeproj
Pod install took 52 [s] to run
Integrating client project
expo_patch_react_imports! took 1.1251 seconds to transform files.
Pod installation complete! There are 106 dependencies from the Podfile and 127 total pods installed.
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

**Comment by cb-palanim** [comment]
@katche70 would you be able to try out something similar to [this](https://github.com/chargebee/chargebee-react-native/issues/178#issuecomment-2113253328)?
